Dockerfile.s390x: notary server does not compile

Building docker on s390x with Dockerfile.s390x currently fails when compiling notary-server with error messages like "cannot find package "github.com/Sirupsen/logrus/hooks/bugsnag"".

See also: (https://jenkins.dockerproject.org/job/Docker%20Master%20%28power,%20z%29%20Binaries/label=z/1315/consoleFull)

**Steps to reproduce the issue:**
1. git clone https://github.com/docker/docker.git
2. cd docker
3. docker build --rm --force-rm -t docker:build -f Dockerfile.s390x .

```
Step 16 : RUN set -x && export GO15VENDOREXPERIMENT=1 && export GOPATH="$(mktemp -d)" && git clone https://github.com/docker/notary.git  "$GOPATH/src/github.com/docker/notary" && (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION") && GOPATH="$GOPATH/src/github.com/docker/notary/vendor:$GOPATH" go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server && rm -rf "$GOPATH"
---> Running in 220ac8c8543f
+ export GO15VENDOREXPERIMENT=1
+ mktemp -d
+ export GOPATH=/tmp/tmp.MmfSLEvL7R
+ git clone https://github.com/docker/notary.git /tmp/tmp.MmfSLEvL7R/src/github.com/docker/notary
Cloning into '/tmp/tmp.MmfSLEvL7R/src/github.com/docker/notary'...
+ cd /tmp/tmp.MmfSLEvL7R/src/github.com/docker/notary
+ git checkout -q docker-v1.11-3
+ GOPATH=/tmp/tmp.43uItm940a/src/github.com/docker/notary/vendor:/tmp/tmp.43uItm940a go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server                                                                                                                              
tmp/tmp.43uItm940a/src/github.com/docker/notary/cmd/notary-server/config.go:9:2: cannot find package "github.com/Sirupsen/logrus" in any of:       

tmp/tmp.43uItm940a/src/github.com/docker/notary/utils/configuration.go:12:2: cannot find package "github.com/Sirupsen/logrus/hooks/bugsnag" in any of:                                                                                                                                                

tmp/tmp.43uItm940a/src/github.com/docker/notary/tuf/data/keys.go:16:2: cannot find package "github.com/agl/ed25519" in any of:                     
```
The reason seems to be that since version "docker-v1.11-1" the notary git repository contains a vendor directory that contains all dependencies.

The following commit has done this:

```
commit 51dc1747e4ab5738039fc21f8a082cba5a6e4f78
Author: Ying Li <ying.li@docker.com>
Date:   Tue Mar 8 11:28:04 2016 -0800

    Move the godeps workspace to the vendor directory to be compliant with Go 1.6
    
    Signed-off-by: Ying Li <ying.li@docker.com>
```

Since the vendor sources are directly under "vendors" and not under "vendors/src" go build does not find them.

We use gcc 5.3 in the image:
```
# go version  
go version go1.4.2 gccgo (GCC) 5.3.0 linux/s390x
```

The following workaround patch that adds a "src" softlink solves the problem:

```
diff --git a/Dockerfile.s390x b/Dockerfile.s390x
index 4edd40e..c678968 100644
--- a/Dockerfile.s390x
+++ b/Dockerfile.s390x
@@ -113,7 +113,7 @@ RUN set -x \
        && export GO15VENDOREXPERIMENT=1 \
        && export GOPATH="$(mktemp -d)" \
        && git clone https://github.com/docker/notary.git "$GOPATH/src/github.com/docker/notary" \
-       && (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION") \
+       && (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION" && ln -s . vendor/src) \
        && GOPATH="$GOPATH/src/github.com/docker/notary/vendor:$GOPATH" \
                go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server \
        && rm -rf "$GOPATH"
```
But I assume this can't be the final solution.

Please advise how to correctly fix this problem.

--- a/Dockerfile.s390x
+++ b/Dockerfile.s390x
@@ -113,7 +113,7 @@ RUN set -x \
 	&& export GO15VENDOREXPERIMENT=1 \
 	&& export GOPATH="$(mktemp -d)" \
 	&& git clone https://github.com/docker/notary.git "$GOPATH/src/github.com/docker/notary" \
-	&& (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION") \
+	&& (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION" && ln -s . vendor/src) \
 	&& GOPATH="$GOPATH/src/github.com/docker/notary/vendor:$GOPATH" \
 		go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server \
 	&& rm -rf "$GOPATH"

